scripts/bracket_orders.js

```diff
@@ -45,29 +48,82 @@ const argv = require("yargs")
   )
   .version(false).argv
 
+// returns undefined if the price was not available
+const getDexagPrice = async function (tokenBought, tokenSold) {
+  // dex.ag considers WETH to be the same as ETH and fails when using WETH as token
+  tokenBought = tokenBought == "WETH" ? "ETH" : tokenBought
+  tokenSold = tokenSold == "WETH" ? "ETH" : tokenSold
+  // see https://docs.dex.ag/ for API documentation
+  const url = "https://api-v2.dex.ag/price?from=" + tokenSold + "&to=" + tokenBought + "&fromAmount=1&dex=ag"
+  let price
+  try {
+    const requestResult = await axios.get(url)
+    price = requestResult.data.price
+  } catch (error) {
+    console.log("Warning: unable to retrieve price information on dex.ag. The server returns:")
+    console.log(">", error.response.data.error)
+  }
+  return price
+}
+
+const thresholdPercent = 2
+const isPriceReasonable = async function (exchange, targetTokenId, stableTokenId, price) {
+  const tokenInfo = await fetchTokenInfo(exchange, [targetTokenId, stableTokenId], artifacts)
+  const targetToken = tokenInfo[targetTokenId]
+  const stableToken = tokenInfo[stableTokenId]
+  const dexagPrice = await getDexagPrice(targetToken.symbol, stableToken.symbol)
+  if (dexagPrice === undefined) {
+    console.log("Warning: could not perform price check against dex.ag.")
+    const answer = await promptUser("Continue anyway? [yN] ")
+    if (answer != "y" && answer.toLowerCase() != "yes") {
+      return false
+    }
+  }
+  if (Math.abs(dexagPrice - price) >= thresholdPercent / 100) {
+    console.log("Warning: the chosen price differs by more than", thresholdPercent, "percent from the price found on dex.ag.")
+    console.log("         chosen price:", price, targetToken.symbol, "bought for 1", stableToken.symbol)
+    console.log("         dex.ag price:", dexagPrice, targetToken.symbol, "bought for 1", stableToken.symbol)
+    const answer = await promptUser("Continue anyway? [yN] ")
+    if (answer != "y" && answer.toLowerCase() != "yes") {
+      return false
+    }
+  }
+  return true
+}
+
 module.exports = async callback => {
   try {
+    await BatchExchange.setProvider(web3.currentProvider)
```

Comment:
We are using these lines of code several times in the project. We should make a function for that...